pkg/tls/certificatemanagement/assets.go

```diff
@@ -0,0 +1,90 @@
+package certificatemanagement
+
+import (
+	"crypto/x509"
+	"fmt"
+	"strings"
+
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/dns"
+)
+
+const (
+	// PrometheusServerTLSSecretName is the TLS secret used when the authentication proxy container in tigera-prometheus
+	// is acting as a server and handles incoming requests.
+	PrometheusServerTLSSecretName = "tigera-prometheus-server-tls"
+)
+
+// TLSAsset should be instantiated at startup only, none of its members are exposed as they should be immutable.
+type TLSAsset struct {
+	secretName         string
+	serviceaccountName string
+	namespace          string
+	permittedDNSNames  []string
+	keyExtUsages       []x509.ExtKeyUsage
+}
+
+func (t TLSAsset) SecretName() string {
+	return t.secretName
+}
+
+func (t TLSAsset) DNSNames() []string {
+	return t.permittedDNSNames
+}
+
+func (t TLSAsset) ExtKeyUsages() []x509.ExtKeyUsage {
+	return t.keyExtUsages
+}
+
+// TLSAssetMap can look up all assets associated with a serviceaccount and determine if a CSR is requesting allowed
+// certificate contents.
+type TLSAssetMap map[string]TLSAsset
+
+var (
+	// TLSAssets Stores the secret name as key and asset as value.
+	TLSAssets           = make(TLSAssetMap)
+	PrometheusServerTLS TLSAsset
+)
+
+// ValidRequestor returns true if the TLS may be created by the service account by means of a CSR.
+func (t TLSAssetMap) ValidRequestor(csrName, requestor string) bool {
+	secretName := strings.Split(csrName, ":")[0] // The CSRName is formatted as "<secretName>:<podName>".
+	if asset, found := t[secretName]; found {
+		return fmt.Sprintf("system:serviceaccount:%s:%s", asset.namespace, asset.serviceaccountName) == requestor
+	}
+	return false
+}
+
+// ValidDNSNames returns true if the request only contains permitted dns names for given secretName.
+func (t TLSAssetMap) ValidDNSNames(csrName string, request *x509.CertificateRequest) bool {
+	secretName := strings.Split(csrName, ":")[0] // The CSRName is formatted as "<secretName>:<podName>".
+	if asset, found := t[secretName]; found {
+		// Create this map at runtime, rather than keeping it in memory as CSRs are issued sporadically.
+		permittedNames := make(map[string]bool, len(asset.permittedDNSNames))
+		for _, permitted := range asset.permittedDNSNames {
+			permittedNames[permitted] = true
+		}
+		for _, name := range request.DNSNames {
+			if !permittedNames[name] {
+				return false
+			}
+		}
+		if request.Subject.CommonName != "" && !permittedNames[request.Subject.CommonName] {
+			return false
+		}
+		return true
+	}
+	return false
+}
+
+// InstantiateTLSAssets Called when the operator initializes.
+func InstantiateTLSAssets(clusterDomain string) {
+	PrometheusServerTLS = TLSAsset{
+		PrometheusServerTLSSecretName,
+		"tigera-prometheus",
+		"tigera-prometheus",
+		dns.GetServiceDNSNames("prometheus-http-api", common.TigeraPrometheusNamespace, clusterDomain),
```

Comment:
Seems like we're jumping through an extra hoop here to dynamically build some global variables that should just be `const` in order to incorporate the `clusterDomain` at startup.

Wouldn't it be nicer if we declared this as constants, removed the need for `Initialize`, and then just passed the clusterDomain as an argument to the `DNSNames` function when calling it? 

It would be simple to just do this all as part of `init()` on program load.